File: python/moneysocket/core/message/request/provider.py
# Copyright (c) 2020 Jarret Dyrbye
# Distributed under the MIT software license, see the accompanying
# file LICENSE or http://www.opensource.org/licenses/mit-license.php

import string
import json

from moneysocket.core.message.request.request import MoneysocketRequest

class RequestProvider(MoneysocketRequest):
    MUST_BE_CLEARTEXT = False

    def __init__(self):
        super().__init__("REQUEST_PROVIDER")

    @staticmethod
    def cast_class(msg_dict):
        c = RequestProvider()
        c.update(msg_dict)
        return c

    @staticmethod
    def check_valid_msg_dict(msg_dict):
        return None


MoneysocketRequest.REQUEST_SUBCLASSES['REQUEST_PROVIDER'] = RequestProvider

File: browser/src/js/moneysocket/core/message/request/provider.js
// Copyright (c) 2020 Jarret Dyrbye
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php


const MoneysocketRequest = require('./request.js').MoneysocketRequest;
const Uuid = require('../../../utl/uuid.js').Uuid;
const BinUtl = require('../../../utl/bin.js').BinUtl;

let REQUEST_SUBCLASSES = require('./request.js').REQUEST_SUBCLASSES;


class RequestProvider extends MoneysocketRequest {
    constructor() {
        super("REQUEST_PROVIDER");
    }

    cryptLevel() {
        return "AES";
    }

    static castClass(msg_dict) {
        var c = new RequestProvider();
        Object.keys(msg_dict).forEach(key => {
            c[key] = msg_dict[key];
        });
        return c;
    }

    static checkValidMsgDict(msg_dict) {
        return null;
    }

}

REQUEST_SUBCLASSES['REQUEST_PROVIDER'] = RequestProvider;

exports.RequestProvider = RequestProvider;
